Sources/GRPCHTTP2Core/GRPCStreamStateMachine.swift

```diff
@@ -192,6 +192,23 @@ private enum GRPCStreamStateMachineState {
 
     var inboundMessageBuffer: OneOrManyQueue<[UInt8]>
 
+    /// This transition should only happen on the client-side.
+    /// It can happen if the request times out before the client outbound can be opened.
```

Comment:
It's not just that: it can happen if the stream closes unexpectedly on the client (i.e. an error happens).